bin-src/lib/compress.js

```diff
@@ -0,0 +1,22 @@
+import archiver from 'archiver';
+import fs from 'fs-extra';
+import { join } from 'path';
+import { file as tempFile } from 'tmp-promise';
+
+export default async function makeZipFile(ctx) {
+  const archive = archiver('zip', { zlib: { level: 9 } });
+  const tmp = await tempFile({ postfix: '.zip' });
+  const sink = fs.createWriteStream(null, { fd: tmp.fd });
+  const { paths } = ctx.fileInfo;
+
+  return new Promise((resolve, _) => {
```

Comment:
The [docs](https://github.com/archiverjs/node-archiver#quick-start) suggest handing `archive.on('warning', ...)` and `archive.on('error', ...)`

Reply:
Would also be good to have some debug logging in here.

Reply:
Good callout, I've added those catches. I figured `warning` could get logged since they're non-blocking but I'm opening to throwing an error in those cases.